rust/worker/src/execution/orchestration/metadata.rs

```diff
@@ -556,13 +563,48 @@ impl MetadataQueryOrchestrator {
         println!("Filtering logs and searching metadata segment");
         self.state = ExecutionState::Filter;
 
-        // TODO: Implement filtering and searching metadata segment
-        // for now we just proxy the items through on the request thread
-        // since in the server we disallow where/where document.
-        // When we implement this we can move it to an operator
+        let metdata_segment = self
+            .get_metadata_segment_from_id(self.sysdb.clone(), &self.metadata_segment_id)
+            .await;
+        let metadata_segment = match metdata_segment {
+            Ok(segment) => segment,
+            Err(e) => {
+                self.terminate_with_error(e, ctx);
```

Comment:
Can I be the bad guy and ask to trace errors such as this one? Something along the lines of `tracing::error!(msg)`

Reply:
Done